library/core/src/char/convert.rs

```diff
@@ -23,7 +23,7 @@ pub(super) const fn from_u32(i: u32) -> Option<char> {
 #[must_use]
 pub(super) const unsafe fn from_u32_unchecked(i: u32) -> char {
     // SAFETY: the caller must guarantee that `i` is a valid char value.
-    if cfg!(debug_assertions) { char::from_u32(i).unwrap() } else { unsafe { transmute(i) } }
+    if cfg!(debug_assertions) { char::from_u32(i).const_unwrap() } else { unsafe { transmute(i) } }
```

Comment:
```suggestion
    unsafe {
        assert_unsafe_precondition!(
            "invalid value for `char`",
            (i: u32) => char_try_from_u32(i).is_ok()
        );
        transmute(i)
    }
```

Reply:
I've no idea why that's not done. I'd love for this to be stabilized so I'm trying to change as little as possible to achieve that goal but if libs is ok with it I can update this PR.

Reply:
`assert_unsafe_precondition` is already used in similar const-stable functions (e.g. [`NonNull::new_unchecked`](https://github.com/rust-lang/rust/blob/4d1bd0db7f489b22c6d8aa2385937a95412c015b/library/core/src/ptr/non_null.rs#L218) or [`unreachable_unchecked`](https://github.com/rust-lang/rust/blob/4d1bd0db7f489b22c6d8aa2385937a95412c015b/library/core/src/hint.rs#L104)) so I doubt that it will be problematic.

Reply:
the original author probably wasn't aware or it didn't exist back then. the macro is the right tool for this

Reply:
Ah, in that case I'll update.